data_structures/linked_list/add_nodes.py

```diff
@@ -19,20 +20,40 @@ def __init__(self):
         self.head = None
 
     # method to print the linked list
-    def printLL(self):
+    def printLL(self) -> None:
         temp = self.head
         while temp:
             print(temp.data, end=" ")
             temp = temp.next
 
     # method to push a new node onto the linked list
-    def push(self, newdata):
+    def push(self, newdata: int) -> None:
         newnode = Node(newdata)
         newnode.next = self.head
         self.head = newnode
+        """
+            >>> llist.push(3)
+            >>> llist.push(2)
+            >>> llist.push(1)
+            >>> ll2.push(2)
+            >>> ll2.push(4)
+            >>> ll2.push(8)
+        """
 
     # method to add the nodes
-    def add(self):
+    def add(self) -> None:
+
+        # Test case for carry generation handling
+        """
+        >>> llist.push(9)
+        >>> llist.push(9)
+        >>> llist.push(9)
```

Comment:
So this test shows that we can call LL.push() but it does not really prove that LL.push() did what we expected it to do.  How can we have a test that _proves_ that 9 is in the linked list twice?